### ISSUE TITLE
No Dropbox Import with Lychee 4.0.4

### ISSUE THREAD

**Issue by gueba** [LycheeOrg/Lychee#565]
Hi,
I installed Lychee 4.04 on my webspace [https://gueba.de/Lychee4/](url) by uploading per ftp and running the installation. Everything seems to work fine exept Dropbox Import.
The key is accepted and works with an installation of Lychee 3 at [https://gueba.de/fotos](url)
The browser reacts on "Import from Dropbox" showing the running band on top but there's no other reaction, no popup. Tried Firefox and Edge.
Any suggestions? Maybe a problem with rights?

Thanks in advance

Edit: Version shows 4.04 but database version 4.0.3

**Comment by ildyria** [comment]
Hi, we just released version 4.0.5 but I can already tell you it will not fix your problem.

Additionally, please do not erase the template we provided, the diagnostic part is useful for us to know things such as :
- if you are running on the release
- if you are using the master branch or a personalized one
- what is the commit number you are at
- if you are using `composer --no-dev`
- if you have set `APP_DEBUG` to `true`
- if you are using `production` environment or `local` or `testing` or `dev`...

Those are precious information for us that we may need to figure out what is going on.

I may agree that in your case it is unlikely to be of any impact but this is still helpful most of the time.

I will try to figure out what is going on... when I have the time... :confused: 

> Edit: Version shows 4.04 but database version 4.0.3

This is unrelated, please try to open a new issue if there are two different things to cover. Also this was already mentionned and is fixed with the 4.0.5 :  see here #550

**Comment by ildyria** [comment]
Will be fixed once #568 is merged.

**Comment by gueba** [comment]
Thank you for answering!

I did a bit more testing and will try to explain

**what I did:**

- Downloaded the *.zip file for 4.0.4 from the Releases-page
- Unzipped to disc
- Uploaded per ftp to my site https://gueba.de/Lychee4/public  (webspace hosted by all.inkl.com). The php info is at https://gueba.de/infophp.php
- created database manually
- accessed the installer in /public/  and ran through the installation. The Installation succeeded without any error.
- created account
- set API key for Dropbox (the key works fine in  https://gueba.de/Lychee-v3/)

```
    Diagnostics
    -------
    Info: Latest version of PHP is 7.4
    Error: Database is behind file versions. Please apply the migration.




    System Information
    --------------
    Lychee Version (release):   4.0.4
    DB Version:                 4.0.3
    
    composer install:           --no-dev
    APP_ENV:                    local
    APP_DEBUG:                  false
    
    System:                     Linux
    PHP Version:                7.3
    MySQL Version:              5.7.28-nmm1-log
    
    Imagick:                    1
    Imagick Active:             1
    Imagick Version:            1673
    GD Version:                 bundled (2.1.0 compatible)




    Config Information
    --------------
    version:                    040003
    check_for_updates:          0
    sorting_Photos_col:         takestamp
    sorting_Photos_order:       ASC
    sorting_Albums_col:         max_takestamp
    sorting_Albums_order:       ASC
    imagick:                    1
    skip_duplicates:            0
    small_max_width:            0
    small_max_height:           360
    medium_max_width:           1920
    medium_max_height:          1080
    lang:                       de
    layout:                     1
    image_overlay:              1
    image_overlay_type:         desc
    default_license:            none
    compression_quality:        90
    full_photo:                 1
    delete_imported:            0
    Mod_Frame:                  1
    Mod_Frame_refresh:          30
    thumb_2x:                   1
    small_2x:                   1
    medium_2x:                  1
    landing_page_enable:        0
    landing_owner:              John Smith
    landing_title:              John Smith
    landing_subtitle:           Cats, Dogs & Humans Photography
    landing_facebook:           https://www.facebook.com/JohnSmith
    landing_flickr:             https://www.flickr.com/JohnSmith
    landing_twitter:            https://www.twitter.com/JohnSmith
    landing_instagram:          https://instagram.com/JohnSmith
    landing_youtube:            https://www.youtube.com/JohnSmith
    landing_background:         dist/cat.jpg
    site_title:                 Lychee v4
    site_copyright_enable:      0
    site_copyright_begin:       2019
    site_copyright_end:         2019
    additional_footer_text:     
    display_social_in_gallery:  0
    public_search:              0
    SL_enable:                  0
    SL_for_admin:               0
    public_recent:              0
    recent_age:                 1
    public_starred:             0
    downloadable:               0
    photos_wraparound:          1
    map_display:                1
    zip64:                      1
    map_display_public:         1
    map_provider:               Wikimedia
    force_32bit_ids:            0
    map_include_subalbums:      1
    update_check_every_days:    3
    has_exiftool:               1
    share_button_visible:       0
    import_via_symlink:         0
    has_ffmpeg:                 0
    location_decoding:          1
    location_decoding_timeout:  30
    location_show:              1
    location_show_public:       1
```
**What does not work:**
From any album hitting the Plus and then "Import from Dropbox" shows no reaction except a very small blue flashing line on top (running from right to left). There is no pop-up window, neither info from firefox.

**What I found out so far:**

Every Version from 4.0.0 shows the same problem.
A freshly installed Version 3.2.16 show the same problem, but can be solved by editing the .htaccess file

original:
`<IfModule mod_headers.c>
	# Here we set CSP for Lychee. The first option is fairly forgiving. The second is stricter, but *should* still work.
	# Additionally, you may wish to set up a report-uri.
	Header set Content-Security-Policy "default-src 'self'; style-src 'self' 'unsafe-inline'; script-src 'self'; connect-src 'self' https://lycheeorg.github.io; form-action 'none'; base-uri 'none'; frame-ancestors 'none'; upgrade-insecure-requests; block-all-mixed-content;"
	# Header set Content-Security-Policy "default-src 'none'; style-src 'self'; script-src 'self'; connect-src 'self' https://lycheeorg.github.io; img-src 'self'; media-src 'self'; prefetch-src 'self'; form-action 'none'; base-uri 'none'; frame-ancestors 'none'; upgrade-insecure-requests; block-all-mixed-content;"
</IfModule>`
Commenting out both options does the trick.

I tried to disable csp in `Lychee4/vendor/bepsvpt/secure-headers/config/secure-headers.php` by setting 'custum-csp' to an empty string, but this didn't work.

**Comment by ildyria** [comment]
Yes I know. I found out where the problem was.
It is already fixed in the master branch and will be available in the next released.
You can see the required change here: https://github.com/LycheeOrg/Lychee/commit/69e53a0b535a05e0286a351f5566bb388749e52b

If you change the file `config/secure-headers.php` as described in the commit, it will work fine.
